src/makielayout/blocks/ballcontroller.jl

```diff
@@ -0,0 +1,706 @@
+end
+function BallController(x::Union{GridPosition, GridSubposition}, axis::Union{LScene, Axis3}; kwargs...)
+    BallController(x; axis = axis, float = false, kwargs...)
+end
+
+
+function initialize_block!(controller::BallController; axis::Union{LScene, Axis3}, float::Bool)
+    blockscene = controller.blockscene
+
+    # Handle block detachment (TODO?)
+    if float
+        align = map(blockscene, controller.halign, controller.valign) do halign, valign
+            return Vec2f(halign2num(halign), valign2num(valign))
+        end
+
+        scene_region = map(blockscene,
+            axis.layoutobservables.computedbbox, controller.float_size, align
+        ) do parent_bb, size, align
+            mini = minimum(parent_bb); ws = widths(parent_bb)
+            anchor = mini .+ align .* ws
+            size = minimum(size) # the scene should end up square
+            origin = anchor .- align * size
+            return round_to_IRect2D(Rect2(origin, Vec2f(size)))
+        end
+    else
+        align = Observable(Vec2f(0, 0))
+        scene_region = map(blockscene, controller.layoutobservables.computedbbox) do bb
+            mini = minimum(bb); ws = widths(bb)
+            center = mini + 0.5 * ws
+            w = min(ws[1], ws[2])
+            return round_to_IRect2D(Rect2(center .- 0.5 * w, Vec2(w)))
+        end
+    end
+
+    scene = Scene(blockscene, px_area = scene_region) # clear = false
+
+    # Handle textures (TODO mipmap?)
+    # scene size threshholds for texture swaps (move to mipmap?)
+    selected_texture = RefValue(:none)
+
+    texture = let
+        w, h = widths(scene_region[])
+        tex = if h < controller.low_mid_threshold[]
+            selected_texture[] = :low
+            controller.texture_low[]
+        elseif h < controller.mid_high_threshold[]
+            selected_texture[] = :mid
+            controller.texture_mid[]
+        else
+            selected_texture[] = :high
+            tex = controller.texture_high[]
+        end
+        Observable(tex)
+    end
+
+    on(scene_region) do bb
+        w, h = widths(bb)
+
+        if h < controller.low_mid_threshold[] - 10 && selected_texture[] != :low
+            selected_texture[] = :low
+            texture[] = controller.texture_low[]
+        elseif controller.low_mid_threshold[] + 10 < h < controller.mid_high_threshold[] - 20 && selected_texture[] != :mid
+            selected_texture[] = :mid
+            texture[] = controller.texture_mid[]
+        elseif controller.mid_high_threshold[] + 20 < h && selected_texture[] != :high
+            selected_texture[] = :high
+            texture[] = controller.texture_high[]
+        end
+    end
+
+    # Generate sphere mesh
+    m = uv_normal_mesh(Tesselation(Sphere(Point3f(0), 1f0), 50))
+    mp = mesh!(
+        scene, m, color = map(x -> ShaderAbstractions.Sampler(x, anisotropic = 16f0), texture),
+        transparency = false, fxaa=!false
+    )
```

Comment:
It seems to me that ShaderAbstractions.jl was meant to allow `Sampler(::Observable)` but that fails to update in GLMakie. Perhaps GLAbstractions needs to `import ShaderAbstraction: update!`?

Reply:
That code doesn't work, ignores updates